src/core/Groove.cpp

```diff
@@ -0,0 +1,40 @@
+
+#include <QObject>
+#include <QLabel>
+
+#include "Groove.h"
+#include "lmms_basics.h"
+#include "MidiTime.h"
+#include "Note.h"
+#include "Pattern.h"
+#include "Song.h"
+
+Groove::Groove()
+{
+}
+
+/*
+ * Default groove is no groove. Not even a wiggle.
+ * @return 0 or -1
+ */
+int Groove::isInTick(MidiTime * _cur_start, fpp_t _frames, f_cnt_t _offset,
+						 Note * _n, Pattern * _p ) {
+
+	return _n->pos().getTicks() == _cur_start->getTicks() ? 0 : -1;
```

Comment:
Why return 0 and -1 instead of a bool?

Reply:
Nevermind - I see in the other `isInTick` methods you're returning integers, and using `-1` as an erorr type.